src/Microsoft.AspNetCore.Mvc.ViewFeatures/Resources.resx

```diff
@@ -131,7 +131,7 @@
     <value>Method '{0}' of view component '{1}' should be declared to return a value.</value>
   </data>
   <data name="ViewComponent_CannotFindComponent" xml:space="preserve">
-    <value>A view component named '{0}' could not be found.</value>
+    <value>A view component named '{0}' could not be found. View components must be a public concrete non-generic class, and should be decorated with ViewComponentAttribute or the class name should end in 'ViewComponent'. The class must not be decorated with NonViewComponentAttribute.</value>
```

Comment:
`View components should be a public concrete non-generic class, and should be decorated with {1} or have a class name ending in '{2}'. The class cannot be decorated with {3}.`